File: src/main/java/iot/tetracube/data/entities/Device.java
package iot.tetracube.data.entities;

import io.quarkus.mongodb.panache.reactive.ReactivePanacheMongoEntity;

import java.util.List;
import java.util.UUID;

public class Device extends ReactivePanacheMongoEntity {

    public UUID deviceId;
    public String name;
    public Boolean isOnline;
    public String alexaSlotId;
    public String hostName;
    public List<DeviceAction> actions;

    public UUID getDeviceId() {
        return deviceId;
    }

    public String getName() {
        return name;
    }

    public Boolean getOnline() {
        return isOnline;
    }

    public String getAlexaSlotId() {
        return alexaSlotId;
    }

    public String getHostName() {
        return hostName;
    }

    public List<DeviceAction> getActions() {
        return actions;
    }
}

File: src/main/kotlin/red/tetracube/smarthub/data/entities/Device.kt
package red.tetracube.smarthub.data.entities

import java.util.*

data class Device(
    val id: UUID,
    val name: String,
    val isOnline: Boolean,
    val feedbackTopic: String,
    val colorCode: String,
    val features: List<Feature> = emptyList()
)